.github/workflows/python-packaging.yml

```diff
@@ -100,3 +117,42 @@ jobs:
     - name: Smoke test ${{ matrix.os }}
       run: python -c 'import tiledbsoma; print(tiledbsoma.libtiledbsoma.__file__)'
       # TODO: more-thorough test
+
+  # ----------------------------------------------------------------
+  publish-to-test-pypi:
+    name: Publish package to TestPyPI
+    needs: wheels
+
+    if: matrix.os == 'ubuntu-22.04' && github.event_name == 'release'
+
+    uses: pypa/gh-action-pypi-publish@master
+    continue-on-error: true
+    with:
+      repository_url: https://test.pypi.org/legacy/
+      user: __token__
+      password: ${{ secrets.TEST_PYPI_TOKEN }}
+      packages_dir: sdist, dist-wheel
+      verbose: true
+
+  # ----------------------------------------------------------------
+  publish-to-pypi:
+    name: Publish package to PyPI
+    needs: wheels
+
+    # Until general release of the main branch of tiledbsoma we'll follow the convention of tagging releases
+    # 0.5.0a1, 0.5.1a1, etc -- always with the "a1" suffix -- that way PyPI will automagically make these
+    # "prereleases". Then:
+    # pip install tiledbsoma -> 0.1.12 (or whatever the main-old branch is at)
+    # pip install --pre tiledbsoma -> 0.5.0a1 (or whatever the main branch is at)
+
+    # For prime time, once we're in 1.0 release:
+    #if: matrix.os == 'ubuntu-22.04' && github.event_name == 'release' && !github.event.release.prerelease
```

Comment:
Once we do release, I recommend still uploading prerelease packages, so users can install them with `pip install --pre`.

Reply:
I thought PyPI did this automagically with the `a` in the release name -- PyPI made it a pre-release even if it wasn't a pre-release at the GitHub level?

* https://github.com/single-cell-data/TileDB-SOMA/wiki#branches-and-releases
* When I made [0.5.0a3](https://github.com/single-cell-data/TileDB-SOMA/releases/tag/0.5.0a3) today I didn't mark it as pre-release at the GitHub-tag level and I was able to install it later with `pip install --pre tiledbsoma`

Reply:
Or oh wait I get it now, nvm -- I'll update the comment

Reply:
@thetorpedodog 
https://github.com/single-cell-data/TileDB-SOMA/pull/701/commits/6cced85663a648fe34c549420515573f9f82133e